### PR TITLE
Added DB Migrations 

### PR DESCRIPTION
**Pull request by RetroSabian**

### CODE REVIEW

File: Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Migrations/MyDbContextModelSnapshot.cs [hunk 1]
```diff
@@ -18,21 +19,383 @@ protected override void BuildModel(ModelBuilder modelBuilder)
...
-            modelBuilder.Entity("Know_Your_Nation_Speedy.Models.Entry", b =>
+            modelBuilder.Entity("Know_Your_Nation_Speedy.Models.Animations", b =>
                 {
-                    b.Property<int>("Id")
+                    b.Property<int>("AnimationsId")
+                        .ValueGeneratedOnAdd()
+                        .HasAnnotation("SqlServer:ValueGenerationStrategy", SqlServerValueGenerationStrategy.IdentityColumn);
+
+                    b.Property<string>("CoverImageLocation");
+
```
Comment (FredJEhlers) [inline Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Migrations/MyDbContextModelSnapshot.cs]: Please remove these extra empty lines

File: Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Animations.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    public class Animations
+    {
+        public int AnimationsId { get; set; }
+        public string Name { get; set; }
+        public string FileLocation { get; set; }
+        public string CoverImageLocation { get; set; }
+        public string Description { get; set; }
+        public ICollection<AnimationsWatched> AnimationWatched { get; set; }
+    }
+}
```
Comment (FredJEhlers) [inline Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Animations.cs]: Most of your model files need an empty line at the end of the file

File: Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Migrations/MyDbContextModelSnapshot.cs (hunk 1 above)
Comment (RetroSabian) [reply Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Migrations/MyDbContextModelSnapshot.cs]: This is an auto generated file

File: Know-Your-Nation-Speedy/NUnitTestKYNS/UnitTest.cs [hunk 3]
```diff
@@ -6,98 +6,76 @@
...
-            public void MockTest(string email)
-            {
-                DbController Obj = new DbController();
-
-                Mock<DbController> chk = new Mock<DbController>();
-                chk.Setup(x => x.DoesEmailExist(Obj, email)).Returns(true);
+            Assert.AreEqual(input,result.Count);
+        }
 
+        /*[TestCase]
```
Comment (FredJEhlers) [inline Know-Your-Nation-Speedy/NUnitTestKYNS/UnitTest.cs:45]: Please remove Ghost Code that is no longer being used or add a `TODO: with a reason why it needs to stay`

File: Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Animations.cs (hunk 2 above)
Comment (RetroSabian) [reply Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Animations.cs]: insert a line at 16? between last field and the } ?

File: Know-Your-Nation-Speedy/NUnitTestKYNS/UnitTest.cs (hunk 3 above)
Comment (RetroSabian) [reply Know-Your-Nation-Speedy/NUnitTestKYNS/UnitTest.cs:45]: Unit tests will be redone once DB migrations are done

**Review by FredJEhlers — APPROVED** [review]

**Review by FredJEhlers — APPROVED** [review]

File: Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/ProductOrders.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+namespace Know_Your_Nation_Speedy.Models
+{
+    public class ProductOrders
+    {
+        public int OrdersId { get; set; }
+        public int ProductsId { get; set; }
+        public Orders Order { get; set; }
```
Comment (guandutoit) [inline Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/ProductOrders.cs:11]: keep your relationships and columns seperated

File: Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Users.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        public int UsersId { get; set; }
+        public string Name { get; set; }
+        public string Surname { get; set; }
+        public string Email { get; set; }
+        public string PhoneNumber { get; set; }
+        public string Password { get; set; }
+        public string MembershipType { get; set; }
+
+        public ICollection<ArticlesRead> ArticleRead { get; set; }
```
Comment (guandutoit) [inline Know-Your-Nation-Speedy/Know-Your-Nation-Speedy/Models/Users.cs:19]: you really got to initialize these here is an example 


```
User(){
Initialise();
}

public void Initialise()
        {
            ArticleRead = new HashSet<ArticleRead >();you can add all the collections in here and just call initialize in the constructor
        }
```

**Pull request merged**